github-email-notifications/githubconsumer.py

```diff
@@ -1,40 +1,148 @@
+        print(
+            "{comment_author} commented on a pull request".format(**comment),
+            file=output
+        )
+        print(comment['comment_body'], file=output)
+        print(
+            "See the full comment at {comment_url}\n".format(**comment),
+            file=output
+        )
+        res = output.getvalue()
+        output.close()
+        return res
+
+    @abstractmethod
     def fmt_pr(self, pull_req):
-        # this could be mail subject
-        print "{pr_author}'s pull request #{pr_num}: \"{pr_title}\" was {pr_action}\n".format(**pull_req)
-        # this could be mail body
+        output = StringIO.StringIO()
+        print(
+            "{pr_author}'s pull request #{pr_num}: \"{pr_title}\" was "
+            "{pr_action}\n".format(**pull_req),
+            file=output
+        )
         if pull_req['pr_action'] == u'opened':
-            print "PR body:\n{pr_body}\n".format(**pull_req)
-        print "See the full pull-request at {pr_url}\n".format(**pull_req)
+            print("PR body:\n{pr_body}\n".format(**pull_req), file=output)
+        print(
+            "See the full pull-request at {pr_url}\n".format(**pull_req),
+            file=output
+        )
+        res = output.getvalue()
+        output.close()
+        return res
 
-class RawPPFormatter(object):
+
+class StdoutFormatter(Formatter):
+    def fmt_issue_comment(self, comment):
+        print(super(StdoutFormatter, self).fmt_issue_comment(comment))
+
+    def fmt_pr(self, pull_req):
+        print(super(StdoutFormatter, self).fmt_pr(pull_req))
+
+
+class EmailFormatter(Formatter):
+    def __init__(self, to_addr, from_addr, smtp_server):
+        """
+
+        :param to_addr: email destinations
+        :param from_addr: source email address
+        """
+        super(EmailFormatter, self).__init__()
+
+        self.to_addr = to_addr
+        self.from_addr = from_addr
+        self.smtp_server = smtp_server
+
+    def _send_email(self, subject, body, attachment=None, new=False):
+        """
+        Inspired by: https://github.com/abartlet/gh-mailinglist-notifications/blob/master/gh-mailinglist.py
+        :param subject:
+        :param body:
+        :param attachment:
+        :param new:
+        :return:
+        """
+        subject = subject.replace('\n', ' ').replace('\r', ' ')
+        outer = MIMEMultipart()
+        outer['Subject'] = Header(subject, 'utf8')
+        outer['To'] = self.to_addr
+        outer['From'] = self.from_addr
+
+        # FIXME messageid for threads
+        #        msgid = "<gh-mailinglist-notifications-%s-%s-%d@%s>" % (
+        #            uuid, repo, pr_number, msg_id_domain)
+        #        if new:
+        #            outer.add_header("Message-ID", msgid)
+        #        else:
+        #            outer.add_header("In-Reply-To", msgid)
+        #            outer.add_header("References", msgid)
+
+        # FIXME attachment (patches, CI output?)
```

Comment:
For the patches you can just include a link to the pull request and append ".patch". For this one it would be https://github.com/freeipa/freeipa-tools/pull/4.patch. I'm not sure we need to attach the patches, though, let's see if anyone requests them.. I think it's easier to just apply the pull request locally with "hub am https://github.com/freeipa/freeipa-tools/pull/4"